datastore/filter.go

```diff
@@ -14,3 +14,9 @@ type SourceFilter struct {
 	Type     string
 	Provider string
 }
+
+type SearchFilter struct {
+	Query    string
+	FilterBy   string
```

Comment:
@jirevwe By how much, we are doing `filterByBuilder` across it might be helpful to pack it into this type. I think we will have testing gains! ✊🏽

Reply:
I don't understand

Reply:
We are doing that task repetitively. I think we can abstract it into that type and test it. Should we huddle?